Amazon/Marketplace/v7153/templates/jfrog-artifactory-ec2-instance.template.yaml

```diff
@@ -0,0 +1,408 @@
+    CreationPolicy:
+      ResourceSignal:
+        Count: !Ref MinScalingNodes
+        Timeout: PT60M
+
+  ArtifactoryLaunchConfiguration:
+    Type: 'AWS::AutoScaling::LaunchConfiguration'
+    Metadata:
+      AWS::CloudFormation::Authentication:
+        S3AccessCreds:
+          type: S3
+          roleName:
+            - !Ref HostRole  # !Ref ArtifactoryHostRole
+          buckets:
+            - !Ref QsS3BucketName
+      AWS::CloudFormation::Init:
+        configSets:
+          artifactory_install:
+            - "config-cloudwatch"
+            - "config-artifactory-primary"
+            - "secure-artifactory"
+        config-cloudwatch:
+          files:
+            /root/cloudwatch.conf:
+              content: |
+                [general]
+                state_file = /var/awslogs/state/agent-state
+
+                [/var/log/messages]
+                file = /var/log/messages
+                log_group_name = /artifactory/instances/{instance_id}
+                log_stream_name = /var/log/messages/
+                datetime_format = %b %d %H:%M:%S
+
+                [/var/log/jfrog-ami-setup.log]
+                file = /var/log/messages
+                log_group_name = /artifactory/instances/{instance_id}
+                log_stream_name = /var/log/jfrog-ami-setup.log
+                datetime_format = %b %d %H:%M:%S
+
+                [/var/log/jfrog-ami-artifactory.log]
+                file = /var/log/messages
+                log_group_name = /artifactory/instances/{instance_id}
+                log_stream_name = /var/log/jfrog-ami-artifactory.log
+                datetime_format = %b %d %H:%M:%S
+              mode: "0400"
+        config-artifactory-primary:
+          files:
+            /root/attach_volume.sh:
+              content: !Sub |
+                #!/usr/bin/env bash
+                IS_PRIMARY="${ArtifactoryPrimary}"
+
+                if [[ $IS_PRIMARY != "true" ]]; then
+                  echo 'Not primary node. Skipping EBS volume attachment.'
+                  lsblk # debug
+                  exit 0
+                fi
+
+                echo "Using primary volume ID ${PrimaryVolume}"
+                VOLUME_ID="${PrimaryVolume}"
+                echo "VOLUME_ID: $VOLUME_ID"
+                if [[ -z "$VOLUME_ID" ]]; then
+                  echo 'Invalid $VOLUME_ID'
+                  exit 1
+                fi
+
+                # Get instance id from AWS
+                INSTANCE_ID=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
+
+                # Attach the volume created by another CFT
+                # the device name should become /dev/nvme1n1
+                # See: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html
+                echo "Attaching volume $VOLUME_ID to instance $INSTANCE_ID"
+                /var/awslogs/bin/aws ec2 attach-volume --volume-id $VOLUME_ID --instance-id $INSTANCE_ID --device /dev/xvdf --region ${AWS::Region}
```

Comment:
ALWAYS QUOTE BASH VARIABLED!

`"${INSTANCE_ID}"`

Also, convention is that CAP_LETTERS are for envars not local vars